### PR TITLE
Update DG TaskList

### PR DESCRIPTION
**Pull request by benedict733**
Update the DG documentation from Task -> TaskList

Update use cases for add and clear tasks

Under extension for clear task so that can edit TaskList, included functionality to update task directly

### CODE REVIEW

**Review by WesleyBLDC — APPROVED** [review]
LGTM

**Comment by WesleyBLDC** [comment]
It fails the JAVA CI

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -595,6 +601,47 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**MSS**
+
+1.  User adds task to a person
+2.  SalesPunch updates the tasklist of the person in the contact list
+
+    Use case ends.
+
+**Extensions**
+
+- 1a. User is able to add more types of tasks to a person.
```
Comment (ajjajjajjajj) [inline docs/DeveloperGuide.md:615]: This doesn't reflect the actual behaviour of the code. From my understanding of your implementation, `addtask` does not prompt check if 'the user is able to add more types of tasks'. There are also no checks for whether a stated task description is empty, or invalid etc. So I think there are no extensions to this use case, unless you wish to add a part about requiring a prefix.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -595,6 +601,47 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+- 1a. User is able to add more types of tasks to a person. 
+
+    - 1a1. SalesPunch requests for the correct data.
+    - 1a2. User enters new data.
+    - Steps 1a1-1a2 are repeated until the data entered is correct.
+    - Use case resumes from step 2.
+
+  Use case ends.
+
+**Use case: Edit tasks**
```
Comment (ajjajjajjajj) [inline docs/DeveloperGuide.md:624]: This feature does not exist in our application. Consider removing this from the DG, or move this to Proposed Implementations, but be sure to add in more context and details about the implementation.

**Review by ajjajjajjajj — COMMENTED** [review]
Thanks for adding in documentation – see my comments for suggested changes

**Pull request merged**